### PR TITLE
Add fibonacci compatibility for splitting ncdm.

### PR DESCRIPTION
**Pull request by adrianbayer**
Can now split the ncdm sphere using a fibonacci scheme. 
Added new input parameter, enum, etc in order to let user choose healpix or fibonacci.

### CODE REVIEW

File: libfastpm/thermalvelocity.c [hunk 1]
```diff
@@ -284,16 +294,24 @@ fastpm_ncdm_init_create(
     nid->n_shells = n_shells;
     nid->n_side = n_side;
     nid->lvk = lvk;
-
+    nid->ncdm_sphere_scheme = ncdm_sphere_scheme;
+    
     /* this is the total number of velocity vectors produced */
-    nid->n_split = 12 * n_shells * n_side * n_side;
+    if (ncdm_sphere_scheme == FASTPM_NCDM_SPHERE_HEALPIX){
```
Comment (rainwoodman) [inline libfastpm/thermalvelocity.c]: Perhaps use a switch case -- compiler will complain for unhandled cases.

**Comment by rainwoodman** [comment]
Looks good. One minor suggestion.

The all lower case 'ncdm' looks a bit strange; but all upper probably looks even weirder?

**Comment by adrianbayer** [comment]
Yeah I agree it looks weird, but it's what I've been using elsewhere. I guess I chose this because the ncdm 'stands out' in between the upper case letters. Happy to change if you prefer something else though.

File: libfastpm/thermalvelocity.c (hunk 1 above)
Comment (adrianbayer) [reply libfastpm/thermalvelocity.c]: Makes sense! I'll fix when Cori is back

**Pull request merged**